services/outputhost/outputhost.go

```diff
@@ -507,22 +507,16 @@ func (h *OutputHost) ReceiveMessageBatch(ctx thrift.Context, request *cherami.Re
 	moreResultTimer := common.NewTimer(timeoutTime.Sub(time.Now()))
 	defer moreResultTimer.Stop()
 MORERESULTLOOP:
-	for remaining := count - 1; remaining > 0; {
+	for remaining := count - 1; remaining > 0; remaining-- {
 		select {
-		case msg := <-cgCache.msgsRedeliveryCh:
-			deliver(msg, cgCache.msgCacheRedeliveredCh)
-			remaining--
-		case msg := <-cgCache.msgsCh:
-			deliver(msg, cgCache.msgCacheCh)
-			remaining--
+		case <-moreResultTimer.C:
+			break MORERESULTLOOP
 		default:
```

Comment:
Oh.. sorry I missed it.. i thought you had all three cases on both places.. looks good now!